pulsar/README.md

```diff
@@ -0,0 +1,47 @@
+# Apache Pulsar Initialization Action
+
+Here we've got an initialization action that automates the installation of a single-cluster pulsar instance on bare metal. More detail can be found in the Pulsar documentation here: https://pulsar.apache.org/docs/en/deploy-bare-metal/
+
+## Usage
+
+Specify the .sh script's location when using gcloud to create your dataproc cluster. Pulsar requires Zookeeper, which can be installed by either including it as an optional component (https://cloud.google.com/dataproc/docs/concepts/components/zookeeper), or by creating a HA cluster by specifying --num-masters=3.
+
+e.g.
+```bash
+gcloud dataproc clusters create cluster-name \
+    --region=region \
+    --initialization-actions=gs://init_action_storage/pulsar-initialization-action.sh \
+    --optional-components=ZOOKEEPER \
+    --metadata=pulsar-functions-enabled=true,builtin-connectors-enabled=true,tiered-storage-offloaders-enabled=true... \
+    ... other flags ...
+```
+
+The initialization action also supports the optional installation of builtin connectors and/or tiered storage offloaders, and enabling pulsar functions. To do so, specify appropriate boolean values in the cluster's metadata (https://cloud.google.com/dataproc/docs/concepts/configuring-clusters/metadata).
+
+e.g.
+```bash
+gcloud dataproc clusters create cluster-name \
+    --region=region \
+    --metadata=pulsar-functions-enabled=true,builtin-connectors-enabled=true,tiered-storage-offloaders-enabled=true... \
+    ... other flags ...
+```
+
+The cluster can be tested by publishing a message using pulsar-client. 
```

Comment:
Add more information about the installation, e.g. where is pulsar installed, which services are running on master, which are on workers, default service ports, how to access the web UI, where to check logs.

Reply:
ack.